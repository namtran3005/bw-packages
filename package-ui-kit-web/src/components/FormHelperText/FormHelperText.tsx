import * as React from 'react';

import { FormHelperTextProps as MuiFormHelperTextProps } from '@material-ui/core/FormHelperText';

import { Typography } from '../Typography';

const FormHelperText: React.FunctionComponent<MuiFormHelperTextProps> = (
  props
) => (
  <Typography.Small
    foregroundAlpha={props.disabled ? 0.5 : 1}
    foreground={props.error ? 'error' : 'subtitleText'}
  >
    {props.children}
  </Typography.Small>
);

export { FormHelperText, MuiFormHelperTextProps as FormHelperTextProps };
