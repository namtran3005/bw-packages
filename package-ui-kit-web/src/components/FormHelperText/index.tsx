export { FormHelperText, FormHelperTextProps } from './FormHelperText';
