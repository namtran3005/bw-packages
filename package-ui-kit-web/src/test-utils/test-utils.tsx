import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
