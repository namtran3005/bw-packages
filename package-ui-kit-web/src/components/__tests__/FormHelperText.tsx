import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { FormHelperText, FormHelperTextProps } from '../FormHelperText';

const defaultProps: FormHelperTextProps = {};

describe('FormHelperText', () => {
  it('should render form helper text', () => {
    const { baseElement } = renderWithTestHarness(
      <FormHelperText {...defaultProps}>Description</FormHelperText>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
