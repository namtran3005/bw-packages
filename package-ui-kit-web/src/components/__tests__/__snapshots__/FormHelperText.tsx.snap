// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHelperText should render form helper text 1`] = `
.c0 {
  text-align: left;
}

.c0.c0.c0 {
  font-weight: 400;
  font-size: 0.75rem;
  color: rgba(109,109,134,1);
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c0.c0.c0:hover {
  background: initial;
}

.c0.c0.c0:active {
  background: initial;
}

<body>
  <div>
    <p
      class="MuiTypography-root c0 MuiTypography-body2"
    >
      Description
    </p>
  </div>
</body>
`;
